src/Main.elm

```diff
@@ -1,16 +1,200 @@
+{-| Because of the fixtures, we don't need a full fledged `Html.program`.
+-}
+main : Program Never Model Msg
+main =
+    Html.beginnerProgram
+        { model = init
+        , update = update
+        , view = view
+        }
+
+
+type alias Model =
+    { {- We could use RoseTree and/or a zipper, but I went with the dead-easy
+         way of filtering the dict (or flat list in my first attempt) each time.
+         Might reconsider this if dealing with performance issues!
+
+         ---
+
+         `Nothing` here means the root folder. We also might create a custom
+         union type to make our intentions more clear:
+
+             type Location = Root | Folder Int
+
+         Some helper functions (`map` + `withDefault` or conversion to `Maybe`
+         come to mind) would then be benefitial.
+      -}
+      currentFolder : Maybe Int
+    }
+
+
+type Msg
+    = OpenFolder (Maybe Int)
+
 
-import Data.Audience
-import Data.AudienceFolder
+{-| We could have this data in model, and in a real app we probably would
+(because this data wouldn't be a constant fixture but something fetched from
+the API). But given the constraints, it's nice to factor this out of the model.
 
+---
+
+I went with `Dict Int Audience` approach over a `List` approach - there was some
+`List.filter`ing when I tried that one initially. So, changed that into `Dict`.
 
-{-| Main file of application
 -}
-main : Html msg
-main =
-    Html.text "There will be app soon!"
+audiences : Dict Int Audience
+audiences =
+    Audience.audiencesJSON
+        |> JD.decodeString Audience.audiencesDecoder
+        |> Result.withDefault []
+        |> List.map (\audience -> ( audience.id, audience ))
+        |> Dict.fromList
+
+
+{-| The same as with `audiences`.
+-}
+audienceFolders : Dict Int AudienceFolder
+audienceFolders =
+    AudienceFolder.audienceFoldersJSON
+        |> JD.decodeString AudienceFolder.audienceFoldersDecoder
+        |> Result.withDefault []
+        |> List.map (\audienceFolder -> ( audienceFolder.id, audienceFolder ))
+        |> Dict.fromList
+
+
+init : Model
+init =
+    { currentFolder = Nothing
+    }
+
+
+{-| Could factor each Msg handler into its own function, but it is currently so
+small that it probably wouldn't be worth it. Depends on the team code style.
+-}
+update : Msg -> Model -> Model
+update msg model =
+    case msg of
+        OpenFolder folderId ->
+            { model | currentFolder = folderId }
+
+
+{-| Gives a bit more meaning to the Dict.get, and lets us not use `flip`.
+-}
+findFolder : Int -> Maybe AudienceFolder
+findFolder folderId =
+    audienceFolders
+        |> Dict.get folderId
+
+
+view : Model -> Html Msg
+view model =
+    let
+        {- This filtering would be unnecessary if we parsed the input data into a RoseTree. -}
+        visibleAudiences : List Audience
+        visibleAudiences =
+            audiences
+                |> Dict.filter (\_ audience -> audience.folder == model.currentFolder)
```

Comment:
`always << (==) model.currentFolder << .folder`
It's just my obsession  :)

Reply:
Yeah, @turboMaCk was telling me about GWI writing in pointfree style a lot :)

Reply:
especially @janjelinek fall for it :D anyway I think in cases like this one it's just matter of personal taste. For an experienced eye, both have a clear intention and this doesn't enforce cryptic naming like `thing, a, item...` as the code isn't that generic so it makes sense to keep it as it is in my opinion.

Reply:
True true, I'm becoming composition and pointfree monster. To be honest it didn't help to readability too much in many times, but you know, my brain sees that automatically, lkie that aiblity to raed wrods with reodrered letters :D